core/src/main/resources/error/error-classes.json

```diff
@@ -32,20 +44,36 @@
   "GROUPING_SIZE_LIMIT_EXCEEDED" : {
     "message" : [ "Grouping sets size cannot be greater than %s" ]
   },
+  "GROUP_INDEX_EXCEED_GROUP_COUNT" : {
+    "message" : [ "Regex group count is %s, but the specified group index is %s" ],
+    "sqlState" : "07009"
```

Comment:
I'm not sure if 07009 is right for group index, it seems to apply to descriptor index. Maybe 42000?